opentelemetry-sdk/src/resource/builder.rs

```diff
@@ -0,0 +1,108 @@
+use std::time::Duration;
+
+use opentelemetry::{KeyValue, Value};
+
+use super::{Resource, ResourceDetector, SERVICE_NAME};
+
+/// Builder to allow easy composition of a Resource
+#[derive(Debug, Default)]
+pub struct ResourceBuilder {
+    resource: Resource,
+}
+
+impl ResourceBuilder {
+    /// Create ResourceBuilder with an empty [Resource].
+    pub fn new_empty() -> Self {
```

Comment:
```suggestion
    pub fn new() -> Self {
```
This makes it easier to reason about. Since on `Resource` the new is the same as empty but it accepts key values. Since this is a builder most would assume it's empty.

Reply:
updated